src/libs/ascent/runtimes/flow_filters/ascent_runtime_relay_filters.cpp

```diff
@@ -730,6 +755,112 @@ RelayIOSave::execute()
+    if (protocol == "overlink")
+    {
+        // either users have specified a topology they want to use
+        if (params().has_path("topologies"))
+        {
+            const conduit::Node &tlist = params()["topologies"];
+            const int num_topos = tlist.number_of_children();
+            if (num_topos != 1)
+            {
+                ASCENT_ERROR("relay_io_save Overlink save requires a single topology; " <<
+                             num_topos << " topologies were requested.");
+            }
+        }
+        // or we must check there is only a single topo in their mesh
+        else
+        {
+            Node local_unique_topos;
+            const int num_doms = in->number_of_children();
+            for (int domain_id = 0; domain_id < num_doms; domain_id ++)
+            {
+                const conduit::Node &dom = in->child(domain_id);
+                if (dom.has_child("topologies"))
+                {
+                    for (const auto & topo_name : dom["topologies"].child_names())
+                    {
+                        if (!local_unique_topos.has_child(topo_name))
+                        {
+                            local_unique_topos[topo_name];
+                        }
+                    }
+                }
+            }
+            Node global_topo_names;
+
+            int rank = 0;
+            int root = 0;
+
+#ifdef ASCENT_MPI_ENABLED
+            MPI_Comm mpi_comm = MPI_Comm_f2c(Workspace::default_mpi_comm());
+            MPI_Comm_rank(mpi_comm, &rank);
+
+            relay::mpi::gather_using_schema(local_unique_topos,
+                                            global_topo_names,
+                                            root,
+                                            mpi_comm);
+#else
+            global_topo_names.append().set_external(local_unique_topos);
+#endif
+            int error = 0;
+            std::ostringstream error_oss;
+
+            Node unique_topos;
+            if (rank == root)
+            {
+                const int num_ranks = global_topo_names.number_of_children();
+                for (int rank_id = 0; rank_id < num_ranks; rank_id ++)
+                {
+                    const conduit::Node &rank_topo_names = global_topo_names.child(rank_id);
+                    for (const auto & topo_name : rank_topo_names.child_names())
+                    {
+                        if (!unique_topos.has_child(topo_name))
+                        {
+                            unique_topos[topo_name];
+                        }
+                    }
+                }
+                const int num_topos = unique_topos.number_of_children();
+                if (num_topos != 1)
+                {
+                    error = 1;
+                    error_oss << "relay_io_save Overlink save requires a single topology; "
+                                 "there are " << num_topos << " topologies in the input mesh.";
```

Comment:
for the error case, lets show them all the unique topology names as part of the error message, and also tell them the param name that they can use to select the topology.

Reply:
How does this look?
```
[Error] Ascent::execute 
file: /usr/workspace/justin/ascent_builds/develop_08_22_24/ascent/src/libs/ascent/runtimes/flow_filters/ascent_runtime_relay_filters.cpp
line: 872
message: 
relay_io_save Overlink save requires a single topology; there are 3 topologies in the input mesh. The current topologies are 
 - topo
 - topo2
 - topo3
You can select which topologies to save using the "topologies" parameter like in the following example:
extracts["<extract_name>/params/topologies"].append() = "<topo_name>"
```